src/icp_canister_backend/tests/rewards_tests.rs

```diff
@@ -159,4 +180,684 @@ fn test_reward_distribution_middle_and_final() {
+        decode_one(&empty_withdrawal).unwrap();
+    assert!(
+        result.is_err(),
+        "Expected InsufficientBalance error for empty withdrawal, got: {:?}",
+        result
+    );
+
+    let valid_withdrawal = pic
+        .update_call(
+            canister_id,
+            user_a,
+            "withdraw_rewards",
+            encode_args((vec![0u64],)).unwrap(),
+        )
+        .expect("Valid withdrawal should succeed");
+
+    let result: Result<Nat, icp_canister_backend::PoolError> =
+        decode_one(&valid_withdrawal).unwrap();
+    let withdrawn_amount = result.expect("Valid user should be able to withdraw their own rewards");
+
+    //user A gets half since there are 2 equal deposits
+    let expected_withdrawal = reward_amount.clone() / Nat::from(2u64);
+    assert_with_error!(
+        &withdrawn_amount,
+        &expected_withdrawal,
+        &ALLOWED_ERROR,
+        "Valid withdrawal amount"
+    );
+}
+
+#[test]
+fn test_multiple_reward_pool_additions_cumulative() {
+    let (pic, canister_id, ledger_id) = setup();
+    let user = Principal::from_text("rdmx6-jaaaa-aaaaa-aaadq-cai").unwrap();
+
+    let deposit_amount = Nat::from(100_000_000u64); // 1 BTC
+    let first_reward = Nat::from(50_000_000u64); // 0.5 BTC
+    let second_reward = Nat::from(30_000_000u64); // 0.3 BTC
+    let third_reward = Nat::from(20_000_000u64); // 0.2 BTC
+    let total_expected_rewards =
+        first_reward.clone() + second_reward.clone() + third_reward.clone(); // 1 BTC total
+
+    let stakable_episode = get_stakable_episode(&pic, canister_id, 7);
+    create_deposit(
+        &pic,
+        canister_id,
+        ledger_id,
+        user,
+        deposit_amount.clone(),
+        stakable_episode,
+    );
+
+    let first_reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+
+    // Add all three rewards
+    reward_pool(&pic, canister_id, ledger_id, user, first_reward.clone())
+        .expect("First reward pool should succeed");
+
+    advance_time(&pic, EPISODE_DURATION / 4);
+    let second_reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    reward_pool(&pic, canister_id, ledger_id, user, second_reward.clone())
+        .expect("Second reward pool should succeed");
+
+    advance_time(&pic, EPISODE_DURATION / 4);
+    let third_reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    reward_pool(&pic, canister_id, ledger_id, user, third_reward.clone())
+        .expect("Third reward pool should succeed");
+
+    let first_end =
+        ((first_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+    let second_end =
+        ((second_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+    let third_end =
+        ((third_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+
+    let latest_end = first_end.max(second_end).max(third_end);
```

Comment:
Why do you need it
You already know the latest end since the test has not dynamic elements.